src/lib.rs

```diff
@@ -402,36 +416,22 @@ pub fn hselect<E, P, A>(
 
 /// An alternative version of homogeneous select, returning the index of the Chan
 /// that is ready to receive.
-#[cfg(feature = "chan_select")]
 pub fn iselect<E, P, A>(chans: &Vec<Chan<E, Recv<A, P>>>) -> usize {
     let mut map = HashMap::new();
 
     let id = {
-        let sel = Select::new();
+        let mut sel = Select::new();
         let mut handles = Vec::with_capacity(chans.len()); // collect all the handles
 
         for (i, chan) in chans.iter().enumerate() {
             let &Chan(_, ref rx, _) = chan;
-            let handle = sel.handle(rx);
-            map.insert(handle.id(), i);
+            let handle = sel.recv(rx);
+            map.insert(handle, i);
             handles.push(handle);
         }
 
-        for handle in handles.iter_mut() {
-            // Add
-            unsafe {
-                handle.add();
-            }
-        }
+        let id = sel.ready();
 
-        let id = sel.wait();
-
-        for handle in handles.iter_mut() {
-            // Clean up
-            unsafe {
-                handle.remove();
```

Comment:
It's gonna be nice to get rid of this :+1: It was always a bit yucky :stuck_out_tongue: